object_detection/detectron2_training.ipynb

```diff
@@ -0,0 +1,243 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "cfg = get_cfg()\n",
+    "cfg.merge_from_file(model_zoo.get_config_file(\"COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml\"))\n",
+    "cfg.DATASETS.TRAIN = (\"my_dataset_train\",)\n",
+    "cfg.DATASETS.TEST = (\"my_dataset_val\",)\n",
+    "cfg.DATALOADER.NUM_WORKERS = 2\n",
+    "cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(\"COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml\")  # Let training initialize from model zoo\n",
+    "cfg.SOLVER.IMS_PER_BATCH = 2  # This is the real \"batch size\" commonly known to deep learning people\n",
+    "cfg.SOLVER.BASE_LR = 0.00025  # pick a good LR\n",
+    "cfg.SOLVER.MAX_ITER = 30000    # \n",
+    "cfg.SOLVER.STEPS = []        # do not decay learning rate\n",
+    "cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128   # The \"RoIHead batch size\". \n",
+    "cfg.MODEL.ROI_HEADS.NUM_CLASSES = 5  # only 5 classes [\"car\", \"chair\", \"cup\", \"person\", and \"traffic light\"] \n",
+    "os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train the model\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "trainer = DefaultTrainer(cfg) \n",
+    "trainer.resume_or_load(resume=False)\n",
+    "trainer.train()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inference & evaluation using the trained model\n",
+    "If you wish to load a trained model to run inference"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluator = COCOEvaluator(\"my_dataset_val\", output_dir=\"../output\")\n",
+    "val_loader = build_detection_test_loader(cfg, \"my_dataset_val\")\n",
+    "cfg.MODEL.WEIGHTS = os.path.join(cfg.OUTPUT_DIR, \"model_final.pth\")  # path to the model we just trained\n",
+    "cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.7   # set a custom testing threshold\n",
+    "predictor = DefaultPredictor(cfg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The given code block implements a function \"format_detectron2_predictions\" that converts the output of Detectron2 to a format that can be used by Cleanlab for identifying label errors. The function accepts the predicted instances and the number of classes as inputs. It processes the predicted bounding boxes and scores for each instance, and outputs a list of numpy arrays containing the bounding boxes and scores for each class."
```

Comment:
```suggestion
    "The code below defines a function \"format_detectron2_predictions\" to convert the prediction output of Detectron2 to a format that can be used by Cleanlab for identifying label errors. This function accepts the predicted object instances and the number of classes as inputs. It processes the predicted bounding boxes and prediction-confidence for each instance, and outputs a list of numpy arrays containing the bounding boxes and prediction-confidence for each class."
```

Reply:
is it for each class? I thought it was just for the predicted (i.e. most likely) class

Reply:
the please delete: "for each class" at the end